Godeps/_workspace/src/k8s.io/kubernetes/pkg/kubectl/cmd/run.go

```diff
@@ -148,6 +156,14 @@ func Run(f *cmdutil.Factory, cmdIn io.Reader, cmdOut, cmdErr io.Writer, cmd *cob
 	}
 
 	generatorName := cmdutil.GetFlagString(cmd, "generator")
+	if len(generatorName) == 0 && opts != nil {
```

Comment:
Discovery doesn't find origin resources today. Our defaults exist on every supported version (dc and pod).